src/tasks/Tasks.js

```diff
@@ -0,0 +1,149 @@
+import PageHeader from '../components/PageHeader';
+import AssignmentIcon from '@material-ui/icons/Assignment';
+import TaskCard from '../components/TaskCard';
+import DisplayMap from '../components/DisplayMap';
+
+const useStyles = makeStyles(theme => ({
+  pageContent: {
+    margin: theme.spacing(5),
+    padding: theme.spacing(3)
+  },
+  searchInput: {
+    width: '75%'
+  },
+  newButton: {
+    position: 'absolute',
+    right: '10px'
+  },
+  cardWrapper: {
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    gap: '15px',
+    marginLeft: '38px'
+  },
+}))
+
+export default function Tasks() {
+  const classes = useStyles();
+  const [records, setRecords] = useState(taskService.getAllTasks());
+  const [openPopup, setOpenPopup] = useState(false);
+  const [recordForEdit, setRecordForEdit] = useState(null);
+  const [filterFn, setFilterFn] = useState({ fn: items => items });
+  
+
+ 
+
+   const addOrEdit = (task, resetForm) => {
+    if (task.id == '0') {
+      taskService.insertTask(task);
+    }
+    else {
+      taskService.updateTask(task);
+    }
+    resetForm();
+    setRecordForEdit(null);
+    setOpenPopup(false);
+    setRecords(taskService.getAllTasks());
+  }
+
+  const deleteTask = (task) => {
+    taskService.deleteTask(task);
+    setRecords(taskService.getAllTasks());
+  }
+
+  const onCompleteChange = (taskId, newCompleted) => {
+    const updatedRecords = records.map(task => {
+      if (task.id == taskId) {
+        return { ...task, isCompleted: newCompleted ? 1 : 0 }
+      }
+      return task;
+    })
+
+    setRecords(updatedRecords);
+  }
+
+  const handleSearch = e => {
+    let target = e.target;
+    setFilterFn({
+      fn: items => {
+        if (target.value === "")
+          return items;
+        else
+          return items.filter(x => x.taskName.toLowerCase().includes(target.value))
+      }
+      
+    })
+  }
+
+  const openInPopup = item => {
+    setRecordForEdit(item)
+    setOpenPopup(true);
+  }
+
+  return (
+    <>
+      <PageHeader
+        title='Todo List'
+        subTitle='Your way to complete tasks easily'
+        icon={<AssignmentIcon fontSize='large' />}
+      />
+      <Paper className={classes.pageContent}>
+        <Toolbar>
+          <SearchBox
+            label='Search Tasks'
+            className={classes.searchInput}
+            InputProps={{
+              startAdornment: (<InputAdornment position="start">
+                <Search />
+              </InputAdornment>)
+            }}
+            onChange={handleSearch}
+          />
+          <Button
+            text="Add Task"
+            variant="outlined"
+            startIcon={<AddIcon />}
+            className={classes.newButton}
+            onClick={() => { setOpenPopup(true); setRecordForEdit(null) }}
+          />
+        </Toolbar>
+      </Paper >
+      <div className={classes.cardWrapper}>
+        {
+         taskService.getAllTasks().map(item => (
```

Comment:
This is not good, it will perform the `getAllTasks()` on every render.  
Please create a `useTasks` hook that exposes a `memoized` array of tasks.